kernel/async_channel/src/lib.rs

```diff
@@ -349,14 +370,21 @@ impl <T: Send> Receiver<T> {
     pub fn is_disconnected(&self) -> bool {
         self.channel.is_disconnected()
     }
+    
+    pub fn sender(&self) -> Sender<T> {
+        Sender {
+            channel: self.channel.clone(),
+        }
+    }
 }
 
 
 /// Drop implementation marks the channel state and notifys the `Sender`
 impl<T: Send> Drop for Receiver<T> {
     fn drop(&mut self) {
         // trace!("Dropping the receiver");
-        self.channel.channel_status.store(ChannelStatus::ReceiverDisconnected);
+        // FIXME
+        // self.channel.channel_status.store(ChannelStatus::ReceiverDisconnected);
```

Comment:
They are because we use `async_channel` as an `mpmc`. `async_channel` will stop working as soon as a single sender or receiver is dropped which seems wrong.

Reply:
ah i see, thanks for identifying that. Looks like that was added in #277 (during a paper deadline crunch) so the author was likely only testing it with a single endpoint of each variety. Should be a simple enough fix: only mark the channel as disconnected when the _last_ Sender or Receiver gets dropped, not just any sender or receiver.

Reply:
I filed #703 about this. For now you can leave this change in there as part of this PR.